src/client/src/producer.rs

```diff
@@ -96,82 +195,219 @@ impl TopicProducer {
         skip(self, buffer),
         fields(topic = %self.topic),
     )]
+    #[deprecated(since = "0.6.2", note = "Please use 'send_keyless' instead")]
     pub async fn send_record<B: AsRef<[u8]>>(
         &self,
         buffer: B,
-        partition: i32,
+        _partition: i32,
     ) -> Result<(), FluvioError> {
-        let record = buffer.as_ref();
-        let replica = ReplicaKey::new(&self.topic, partition);
-        debug!("sending records: {} bytes to: {}", record.len(), &replica);
+        let buffer: Vec<u8> = Vec::from(buffer.as_ref());
+        self.send_keyless(buffer).await?;
+        Ok(())
+    }
+}
 
-        let spu_client = self.pool.create_serial_socket(&replica).await?;
+async fn group_by_spu(
+    topic: &str,
+    partitions: &StoreContext<PartitionSpec>,
```

Comment:
doesn't need dependencies to `StoreContext`, it just need `LocalStore` which assume cache are populated

Reply:
The StoreContext is the type with the `lookup_by_key` method which I'm using to implement this. It is not obvious to me how to go about this with LocalStore, and what is written here is working well

Reply:
Actually, this may be good.  If live SPU is not found, it can wait.  Ok. let's leave as is